### PR TITLE
refactor(drift): Parallelize requests using conc

### PR DESCRIPTION
**Pull request by dcreey**

### CODE REVIEW

File: pkg/commands/drift/drift.go [hunk 1]
```diff
@@ -129,34 +159,58 @@ func actualGCPIAM(
...
-	oIAM, err := client.OrganizationIAM(ctx, organizationID)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get organization IAM for ID '%s': %w", organizationID, err)
-	}
-	for _, i := range oIAM {
-		m[URI(i, organizationID, foldersByID, projectsByID)] = i
-	}
-	for _, f := range foldersByID {
-		fIAM, err := client.FolderIAM(ctx, f.ID)
+	p := pool.New().
```
Comment (drevell) [inline pkg/commands/drift/drift.go:162]: If we used NewWithResults / ResultPool, could we avoid the need for channels?

**Review by drevell — CHANGES_REQUESTED** [review]

**Comment by drevell** [comment]
Thanks for trying this out, this is fun.

File: pkg/commands/drift/drift.go (hunk 1 above)
Comment (dcreey) [reply pkg/commands/drift/drift.go]: Done

**Review by drevell — APPROVED** [review]

**Comment by drevell** [comment]
> Should note that using the pool comes with some overhead.
> 
> ```
> // Pool is efficient, but not zero cost. It should not be used for very short
> // tasks. Startup and teardown come with an overhead of around 1µs, and each
> // task has an overhead of around 300ns.
> ```
> 
> https://github.com/sourcegraph/conc/blob/main/pool/pool.go#L27C1-L29C41
> 
> This shouldn't be an issue for something like iam drift detection - but this could be more problematic if you want to write something high performance.

Good to know, thanks for the education.

**Pull request merged**